docs/maintenance/BRANDING.md

```diff
@@ -0,0 +1,37 @@
+---
+id: branding
+title: Branding
+---
+
+This document serves as a guide for how we generally describe or represent typescript-eslint.
+It's fine if you don't always adhere to it.
+
+## Terminology
+
+### Name
+
+Our project name is **_"typescript-eslint"_**.
+We sometimes refer to it as _"tseslint"_ for short.
```

Comment:
bikeshed: ts-eslint vs tseslint

I like the dash because it further distances it from tslint.
Without the dash it seems awfully close and could easily be confused with it at a glance.

Also the dash aligns with our non-shortened form.

Reply:
It's a pity the Twitter account is `@tseslint`. Maybe we should make `@ts-eslint` on Fosstodon just to be confusing 😄